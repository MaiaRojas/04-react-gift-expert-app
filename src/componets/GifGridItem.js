import React from 'react';
import PropTypes from 'prop-types'


export const GifGridItem = ({ title, url}) => { 
  return (
    <div className="card animate__fadeIn">
      <p>{ title }</p>
      <img src={url} alt={title} />
    </div>
  )
};

GifGridItem.protoTypes = {
  title: PropTypes.string.isRequired,

}