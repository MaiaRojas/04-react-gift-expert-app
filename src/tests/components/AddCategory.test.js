import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { AddCategory } from '../../componets/AddCategory';

describe('Pruebas en <AddCategory />', () => {

  const setCategories = jest.fn();
  let wrapper = shallow( <AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow( <AddCategory setCategories={setCategories} />);
  });

  test('should be displayed correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });

  test('should change the text box', () => {

    const input = wrapper.find('input');
    const value = 'Hello world!'

    input.simulate( 'change', { target: { value }});

    expect(wrapper).toMatchSnapshot();
  });

  test('should not called the info with submit', () => {

    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should called setCtegories and clean the text box', () => {
    const input = wrapper.find('input');
    const value = 'Hello world!';

    input.simulate( 'change', { target: { value }});
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    const inputValue = wrapper.find('p');

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(inputValue.text()).toBe('');

  });

});
