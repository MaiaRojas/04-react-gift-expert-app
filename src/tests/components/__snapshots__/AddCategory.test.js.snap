// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> should be displayed correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;

exports[`Pruebas en <AddCategory /> should change the text box 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Hello world!
  </p>
  <input
    onChange={[Function]}
    type="text"
    value="Hello world!"
  />
</form>
`;
